File: functions/updateBudget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, spent, expenses;

float input(char type[], float var){
    printf("monthly %s:\n", type);
    scanf("%f", &var);
    return var;
}

void percent(char type[], int ammount){
    int per = ammount/income * 100;

    printf("your %s is %d%% of your income.\n", type, per);
}
    
int main(void){
    
    printf("this is a budget calculator.\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries);
    transportation = input("transportation", transportation);
    
    savings = income *.2;
    expenses = rent + utilities + groceries + transportation;
    spent = income - expenses - savings;
    
    printf("you make %.2f\n", income);
    printf("your expenses are $%.2f\n", expenses);
    printf("your savings are $%.2f\n", savings);
    printf("your spending money is $%.2f\n", spent);

    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("savings", savings); 
    percent("expenses", expenses);
    percent("spending", spent);   
    return 0;
}

File: functions/updateBudget.py
print("hello, this is a financial calculator!\n")

def input(inputs):
    questions = float(input(f"What is your monthly {inputs}?\n"))
   
    return questions

income= input ("income")
rent = input ("rent")
utilities = input ("utilities ")
groceries = input ("groceries")
transportation = input ("transportation")
saving = income*2
expenses = rent+ utilities + groceries + transportation
spend = income - expenses - saving

def percent(type, amount):
    per = amount/income *100

    return(f"Your {type} is  {per}% income")

print(f"Your monthly income is ${income:.2f}\n")
print(f"Your monthly expenses are ${expenses:.2f}\n")
print(f"Your monthly savings is ${saving:.2f}\n")
print(f"Your monthly spending money is ${spend:.2f}\n")
print(percent("rent",rent))
print(percent("utilities", utilities))
print(percent("groceries", groceries))